src/TLuaInterpreter.cpp

```diff
@@ -10042,101 +10060,131 @@ int TLuaInterpreter::setGridMode(lua_State* L)
 // Documentation: https://wiki.mudlet.org/w/Manual:Lua_Functions#setFgColor
 int TLuaInterpreter::setFgColor(lua_State* L)
 {
-    int s = 1;
+    int s = 0;
     int n = lua_gettop(L);
-    string a1;
+    QString windowName;
     int luaRed;
     int luaGreen;
     int luaBlue;
     if (n > 3) {
-        if (lua_isstring(L, s)) {
-            a1 = lua_tostring(L, s);
-            s++;
+        if (!lua_isstring(L, ++s)) {
+            lua_pushfstring(L, "setFgColor: bad argument #%d type (window name expected, got %s!)", s, luaL_typename(L, s));
```

Comment:
`"...window name as string expected,..."` given that we are reporting the wrong type of argument we need to report the type of argument we **do** want (a string in this case)!

Reply:
applied